src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -1998,10 +1998,10 @@ private IEnumerable<Diagnostic> FreezeDeclarationDiagnostics()
         {
             _declarationDiagnosticsFrozen = true;
 
-            // Also freeze generated attribute flags by observing them
-            // symbols bound after getting the declaration diagnostics shouldn't need to modify the flags
-            _needsGeneratedIsReadOnlyAttribute_IsFrozen = true;
-            _needsGeneratedIsByRefLikeAttribute_IsFrozen = true;
+            // Also freeze generated attribute flags.
+            // Symbols bound after getting the declaration
+            // diagnostics shouldn't need to modify the flags.
+            _needsGeneratedAttributes_IsFrozen = true;
```

Comment:
Not blocking this PR: this raises the follow-up question: are declaration diagnostics and generated attributes getting frozen at the same phase?

Reply:
What did you think about this? Should we only have one "frozen" flag here too?

---
In reply to: [146961651](https://github.com/dotnet/roslyn/pull/22820#discussion_r146961651) [](ancestors = 146961651)

Reply:
Logged #22887.